lcls_tools/common/devices/reader.py

```
@@ -142,6 +143,24 @@ def create_lblm(
         return LBLMCollection(**device_data)
 
 
+def create_bpm(
+    area: str = None, name: str = None
+) -> Union[None, BPM, BPMCollection]:
+    device_data = _device_data(area=area, device_type="bpms", name=name)
+    if not device_data:
+        return None
```

I don't love this pattern in general, though it's not your issue. I feel like `_device_data` should throw an error if it fails to find a YAML file. It leads to situations like this where the caller, `create_bpm`, has to catch a silent failure, then communicate that silent failure to its own caller by returning `None`. The same effect could be achieved if `_device_data` would just throw its own error.

I know there are situations where the caller might want to ignore the fact that a BPM can't be initialized, as in the case of mass BPM initialization. In that case, the caller can just create their own try/catch block.

That being said, adding errors to `_device_data` is probably out of scope for this PR. If people agree with me, I can write an issue and self assign it.

---

I'm in support of the proposed `_device_data` fix